test/type/opaque.swift

```diff
@@ -394,6 +394,6 @@ protocol OpaqueProtocolRequirement {
 
 func testCoercionDiagnostics() {
   var opaque = foo()
-  opaque = bar() // expected-error {{cannot assign value of type 'some P' to type 'some P'}} {{none}}
+  opaque = bar() // expected-error {{cannot assign value of type 'some opaque.P' to type 'some opaque.P'}} {{none}}
```

Comment:
I guess this doesn't have to be addressed in this PR, but qualifying `some P` doesn't make the diagnostic better here because the problem isn't a name collision.

Without attempting to address the specific diagnostic message for opaque types, though, could there be some logic to avoid qualifying opaque types--or for that matter, any types where the resulting strings would be identical?

Reply:
yeah, this is just temporary. @jrose-apple has been working on #25624 which builds on this patch and does pretty much exactly what you suggest.